### ISSUE TITLE
Ghelper didn't work for me.

### ISSUE THREAD

**Issue by aroh100876** [seerge/g-helper#3513]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information or no app logs will be provided, my issue will be closed without an answer.

### What's wrong?

OK, since you asked me to explain the bug here :)

I really didn't have any problems with Armoury Crate, but I found out GHelper at r/Asustuf and since I really like simplicity, decided to try it out.

So, just so you know, I used X's GROK to unistall AC and followed all the steps it gave me including using the official tool from the ASUS website. And then I installed GHelper. But then it was already too late and just turned the computer off.

Yesterday I was watching a Hockey game on the ZenScreen connected to the Thunderbolt 4 (which I use basically for streaming only since I know that the TB4 is connected directly to the iGPU) and decided to play Black Myth Wukong.

Two things happened:

- To run the games I always open the Steam app, go to Library, select the game and click play (I know there are easier ways, but that's how I do it) and it took a lot of time for the game to start. I didn't think much about it at first.
- The FPS on the hockey game went super low, I would say less than 1 FPS. At first I thought it had to do with the website I was watching it from because let's say it was not from an official source.

Anyway, at some point I decided to close the game and see what happened with the Hockey game and it went back to normal. But since I know BMW is a very demanding game (basically the reason I sold my old computer and bought this one) I tried with Ghost of Tshushima and got the two problems again. I closed the Hockey game and opened Netflix In the browser and started streaming something and it had the same FPS problem. I tried both the Standard and Ultimate modes but I got the same results with both of them. I really don't know what the Optimus thing is so I can't say anything about it.

### How to reproduce the bug?

Just start a game.

### Logs

Don't have any, I searched for them but there is none.

### Device and Model

ASUS TUF F15 model FX507VI-F15.I74070

### Additional information.

Intel Core i7-13620H and NVIDIA GeForce RTX 4070 (8 GB). I upgraded the memory to 2x32 GB Corsair Vengeance SODIMM DDR5 RAM 5200 MHz and added a Crucial P5 Plus - SSD NVMe M.2, 1TB PCIe 4.0 3D NAND. On top of that I have two ZenScreens: Asus ZenScreen MB16ACV that is connected to the Thunderbolt 4 port and Asus ZenScreen MB16AHV conected to the HDMI. Also, connected to the USB 3.2 Gen1 Type-A port there is a Razer Tartarus Pro Gaming KeyPad and to the USB 3.2 Gen2 Type-C port there is a WD_BLACK P40 Game Drive SSD 2TB where all my games are installed. On the right side I connected a LLano RGB cooling pad and connected to it there is a Logitech Power Play Mouse Pad.

### Armoury Crate

Uninstalled

### Asus Services

None, I used GHelper to stop all of them.

### Version

The downloadable version from 2 days ago.

### OS

Windows 11 24H2 Build 26100.2605

**Comment by seerge** [comment]
@aroh100876 hello,

- I didn't completely get from your explanation - did you run your games on the external monitor or on internal display of the laptop?
- Did you play plugged in or from battery ?
- Do you have [Game Mode turned on in windows
](https://support.xbox.com/en-US/help/games-apps/game-setup-and-play/use-game-mode-gaming-on-pc) ? 
- What is selected under NVidia Control Panel -> Display Mode ?
![image](https://github.com/user-attachments/assets/7abf58fe-ed1e-4a64-8617-a3427566a2b2)

As for log - you can find app log under `%appdata%\GHelper\` folder (just copy paste this into address bar of explorer) and it will open the folder, then you can just drag-n-drop it into reply.

Thanks

**Comment by aroh100876** [comment]
@seerge 

- I run the games on the laptop's internal display.
- Laptop is always plugged. This laptops eat the battery in minutes.
- Yes, Game Mode is on.
- I don't have the "Manage Display mode" option.

And here is the log :)

[log.txt](https://github.com/user-attachments/files/18232793/log.txt)

**Comment by seerge** [comment]
@aroh100876 ok great,

You don't see Display Mode in Nvidia Control panel cause you are already in Ultimate mode (so display is routed to dGPU all the time). That's fine and how it should be. Optimus is available only in Standard :)

From the log it looks like you were on all default `Turbo` mode without any custom settings on top. The only change is probably an `RTS`, Visual mode (Game Visual in AC). So there is nothing that should be limiting performance :)

Can you do a quick test now:

1. Run your game with AC to make sure it runs ok ?
2. Then without uninstalling anything (for try it's not necessarily) just download latest [release of G-Helper](https://github.com/seerge/g-helper/releases), set it to the same `Turbo` you used, and run a game again ? Does it run same ?

Thanks.

**Comment by aroh100876** [comment]
@seerge sorry this took so long but I spent the whole evening super busy.

So, I did what you said and rand G-Helper with AC still on the computer... and it was a mess. I got again the same two problems. So, I made sure the Run on Startup was NOT checked and restarted the computer and with only AC working and not G-Helper the problems were still there. So I uninstalled AC and took a video of the the whole thing to show it to you.
Then I deleted G-Helper, reinstalled AC and restarted the computer and took a video of the computer working correctly so you can see the difference.

The problem is that the files are too big, so here is a link where you can see both videos if you want.

https://1drv.ms/f/c/906bc4366fd38ada/EkF0oPx51aVDnB_zlXuI-sMByVZEbinzGTa9O8DhwwhOsw?e=We80Rl

You can see on GHelper 1.mp4 that it takes almost a minute for the game to start vs 15 secs on GHelper 2.mp4. You can also see what happens to the movie on the TB4 monitor. I also included two pictures of the G-Helper configuration I had.

Anyway, G-Helper is doing something on my computer that changes stuff around and stays even if G-Helper doesn't start up when starting the computer.

**Comment by seerge** [comment]
@aroh100876 hello, ok. Thanks for checking!

Thing is that G-Helper wouldn't really even change anything in your case besides selecting a baked-in `Turbo` preset + selecting a default windows "Balanced" power plan in Control Panel.

It could be that something in the default power plan in windows was already changed  (that restricts performance).

When you will have time can you: 

1. Run [latest release of G-Helper](https://github.com/seerge/g-helper/releases) in `Turbo` mode again and check if game still starts slowly (no need to uninstall AC)
2. (Just as a sanity check) set `Visual mode` in the app to `Default` (instead of RTS) and check a game again ?
3. Try to reset power plan to [defaults as explained here](https://github.com/seerge/g-helper/wiki/Troubleshooting#resetting-windows-power-plan-to-defaults)
4. Select `Turbo` in the app, then `Fans+Power` section and click `Factory Defaults` there too and check how game runs again?
5. If it doesn't help - open `Fans+Power` and set `Power Mode` (on the left) to `High Performance Plan`. Then check a game again ?

For this you don't need to uninstall AC. Furthermore every time you select a mode in AC or G-Helper they would set it over existing setting.

**Comment by aroh100876** [comment]
@seerge OK, did all you told me:

- The game started faster, 30 secs this time, but is till twice as slow as with AC.
- I still have the FPS problem on the TB4 monitor.

**Comment by seerge** [comment]
@aroh100876 hello, thanks

For can you just quit G-Helper, then manually select Turbo (or whatever you usually use) mode in AC and check how game runs ? 

To clarify : when you set mode in AC it would completely **overwrite** whatever was previously set. So it should run ok?

And can you also upload latest log from G-Helper again ? :) 

Thanks

**Comment by aroh100876** [comment]
@seerge So, I did something different.

- I got CrystalDiskMark to test my External SSD speed and find out that I get the same speeds using The USB 3.2 Gen2 port than the TB4 port (I thought I had read it was not compatible, but what I read is that it wouldn't give me the same speed than with a USB 3.2 Gen2x2 port) so, since I don't have the 2x2 port it's exactly the same.
- Since I get the same speeds, I moved the SSD to the TB4 port and the monitor to the USB 3.2 Gen2 port (had to download some drivers from ASUS to have it working).
- Started the game (with G-Helper running) while streaming on the monitor that was giving me problems and everything went smooth.

So, in conclusion, I think I've solved the problem and figure out what was causing it. Is there any chance that the whole monitor connected to the TB4 was making it slow for the games to start? Like I don't know much about computers but what the average user does. But everything got fixed after I moved the monitor from port.

**Comment by seerge** [comment]
@aroh100876 ah, good find. So your game was running from external SSD? I have kind of missed that in a first place ... (but i see you mentioned in initial post).

I have never thought about it before. But if SSD is slow (no matter why) - games would obviously run slowly from it :) 

It's still strange that it was some differences between AC and G-Helper there ? Since G-Helper can't and doesn't affect SSD speeds at all. My only guess would be that there is some hidden setting in windows default "Balanced" power plan that was affecting external SSD read speeds ? But that would be quite strange if windows does that anyway.

**Comment by aroh100876** [comment]
@seerge I have bad news again:

So, I restarted my computer and the FPS on the... let's call it streaming monitor went down again. But what's even weirder, I moved the video to the HDMI monitor and got the same exact problem. So, it's not the port. Something is happening when I uninstall AC that does it. Sorry man :( I'm sending you the log file again, but other than that I think we've already tried everything. Back to installing AC again :(

[log.txt](https://github.com/user-attachments/files/18256452/log.txt)

**Comment by seerge** [comment]
@aroh100876 ok,

I see that use `RTS` Visual Mode. Can you set Visual mode to `Default` or even `Disabled` to see if it makes difference on your streaming monitor ?

Also did you had a chance to check if different Power Modes affect SSD speed in CrystalDiskMark?

**Comment by aroh100876** [comment]
@seerge didn't work :(

Got the same problem the Visual Mode on Disabled and the Charge Limit at 100%. I have a WoW raid in 10 minutes and I need to get AC installed again and the computer ready. If you have any other idea I'll do it tomorrow. But I think we already did everything it can be done. There is something about AC that keeps the computer working smoothly.

Ah, and by the way, the SSD speed was the same at Best Performance and High Performance.

**Comment by aroh100876** [comment]
@seerge I just found out that I just wasted your time :(

If it's any consolation, I also wasted mine. We didn't get enough people for the raid, it being Xmas time and a very small guild, so I started testing some stuff. I've been using Black Myth Wokong all day to test everything you asked me to test and it happens that BMW is the most demanding game I have, followed by RDR2 and guess what, I got the FPS thing on the streaming monitor when I ran both games with AC. The same problem I was having with G-Helper. So once again, I uninstalled AC, ran G-Helper and tried with less demanding games and... I don't have the FPS problem anymore. I'm and idiot. G-Helper had nothing to do with it.

Sorry man :(

**Comment by seerge** [comment]
@aroh100876 ok, glad to hear you have sorted it out. Do I understand correctly that game runs as you expect it to then ? :)

**Comment by aroh100876** [comment]
@seerge yeah, it's running as intended... I think... I installed, uninstalled, downloaded, deleted AC and G-Helper so many times that all I know is that now games are running the same with both of them lol.

But yeah, thanks a lot for helping me out man :D

**Comment by seerge** [comment]
@aroh100876 ok, great :)

On a side note: I would still move all games that you are constantly playing to the internal SSD as it's most likely faster than one connected over USB :)

**Comment by seerge** [comment]
@aroh100876 hello.

Just one extra idea what could be the reason of your problem in a first place. In windows 11 H24 they have allowed `Energy Saver` in the tray menu to be permanently enabled (before it was possible only when you were on battery).

![Image](https://github.com/user-attachments/assets/fd9a7bca-93a5-4160-b9dd-e7828509b17d)

This mode forces lower power consumption and also restricts some other things (most likely affecting Thunerbolt 4 / USB4 too). 

It could be that Armoury's custom power plan would just force `Energy Saver` to be OFF no matter what, and since G-Helper just uses default power plan from windows (as it's more flexible) - it would keep `Energy Saver` ON. That could be the difference.
----

Do you mind to try if your Thunderbolt 4  setup with running game from external SSD is indeed affected by this mode in windows ?

Thanks!

**Comment by aroh100876** [comment]
Hey, I bet you don't even remember this conversation but I found out (almost by accident) what was causing the low frame rate on the secondary monitors and it had nothing to do with GHelper. The whole thing had to do with the browsers (I tested Firefox, Chrome and Brave) and they all had an option under Settings > System > Use graphics acceleration when available. Leaving that On causes the problem, if you turn it off the problem is gone. An like I said it has nothing to do with GHelper. 
Just wanted you to know because I know you care about your program and you were nice enough to help me, so there it is, if anybody ever has the problem you know what's causing it :)
